components/automate-cli/cmd/chef-automate/sshAutomateHA.go

```diff
@@ -239,57 +258,111 @@ func runSshCommand(cmd *cobra.Command, args []string) error {
+// 	}
+// }
+
+func getAutomateHAInfraDetails(filePath string) (*AutomateHAInfraDetails, error) {
+	automateHAInfraDetails := &AutomateHAInfraDetails{}
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(file)
+
+	content := buf.String()
+	if content == "" {
+		return nil, errors.New("the file is empty")
 	}
+
+	err = json.Unmarshal([]byte(content), automateHAInfraDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	extractPortAndSshUserFromAutomateSSHCommand(automateHAInfraDetails)
+	return automateHAInfraDetails, nil
 }
 
-func extractPortAndSshUserFromAutomateSSHCommand(automateHAInfraDetails *AutomteHAInfraDetails) {
-	if automateHAInfraDetails != nil && len(automateHAInfraDetails.Outputs.AutomateSSH.Value) > 0 {
-		automateSSH := automateHAInfraDetails.Outputs.AutomateSSH.Value[0]
-		lastspaceIndex := strings.LastIndex(automateSSH, " ")
-		if len(strings.TrimSpace(automateHAInfraDetails.Outputs.SSHPort.Value)) < 1 {
-			if strings.Contains(automateSSH, "-p") {
-				port := strings.TrimSpace(automateSSH[strings.LastIndex(automateSSH, "-p")+2 : lastspaceIndex])
-				automateHAInfraDetails.Outputs.SSHPort.Value = port
-			} else {
-				automateHAInfraDetails.Outputs.SSHPort.Value = "22"
-			}
-		}
-		if len(strings.TrimSpace(automateHAInfraDetails.Outputs.SSHUser.Value)) < 1 {
-			sshUser := strings.TrimSpace(automateSSH[lastspaceIndex:strings.LastIndex(automateSSH, "@")])
-			automateHAInfraDetails.Outputs.SSHUser.Value = sshUser
-		}
+// TODO: restructure
+// func extractPortAndSshUserFromAutomateSSHCommand(automateHAInfraDetails *AutomateHAInfraDetails) {
+// 	if automateHAInfraDetails != nil && len(automateHAInfraDetails.Outputs.AutomateSSH.Value) > 0 {
+// 		automateSSH := automateHAInfraDetails.Outputs.AutomateSSH.Value[0]
+// 		lastspaceIndex := strings.LastIndex(automateSSH, " ")
+// 		if len(strings.TrimSpace(automateHAInfraDetails.Outputs.SSHPort.Value)) < 1 {
+// 			if strings.Contains(automateSSH, "-p") {
+// 				port := strings.TrimSpace(automateSSH[strings.LastIndex(automateSSH, "-p")+2 : lastspaceIndex])
+// 				automateHAInfraDetails.Outputs.SSHPort.Value = port
+// 			} else {
+// 				automateHAInfraDetails.Outputs.SSHPort.Value = "22"
+// 			}
+// 		}
+// 		if len(strings.TrimSpace(automateHAInfraDetails.Outputs.SSHUser.Value)) < 1 {
+// 			sshUser := strings.TrimSpace(automateSSH[lastspaceIndex:strings.LastIndex(automateSSH, "@")])
+// 			automateHAInfraDetails.Outputs.SSHUser.Value = sshUser
+// 		}
+// 	}
+// }
+
+func extractPortAndSshUserFromAutomateSSHCommand(automateHAInfraDetails *AutomateHAInfraDetails) {
+	if automateHAInfraDetails == nil || len(automateHAInfraDetails.Outputs.AutomateSSH.Value) == 0 {
+		return
+	}
+
+	automateSSH := automateHAInfraDetails.Outputs.AutomateSSH.Value[0]
+	lastspaceIndex := strings.LastIndex(automateSSH, " ")
+
+	if len(strings.TrimSpace(automateHAInfraDetails.Outputs.SSHPort.Value)) > 0 {
```

Comment:
This logic is not correct. We cannot return only if Port is available. The logic in the original function was to get user and port value from other string if those data are missing in the structure fields